parse.go

```diff
@@ -240,3 +249,52 @@ func base64urlTrailingPadding(s string) ([]byte, error) {
 	s = strings.TrimRight(s, "=")
 	return base64.RawURLEncoding.DecodeString(s)
 }
+
+// Strips eventual optional members of a JWK in order to be able to compute the thumbprint of it
+// https://datatracker.ietf.org/doc/html/rfc7638#section-3.2
+func getThumbprintableJwkJSONbytes(jwk map[string]interface{}) ([]byte, error) {
+	minimalJwk, err := parseJwk(jwk)
+	if err != nil {
+		return nil, err
+	}
+	jwkHeaderJSONBytes, err := getKeyStringRepresentation(minimalJwk)
+	if err != nil {
+		return nil, err
+	}
+	return jwkHeaderJSONBytes, nil
+}
+
+// Returns the string representation of a key in JSON format.
+func getKeyStringRepresentation(key interface{}) ([]byte, error) {
+	var keyParts interface{}
+	switch key.(type) {
+	case *ecdsa.PublicKey:
```

Comment:
You can assign key in the case which removes the need to type cast inside the case. 
```suggestion
	case key = *ecdsa.PublicKey:
```

Reply:
Done. Did not know that. Was very elegant